src/components/SearchForm.js

```diff
@@ -0,0 +1,74 @@
+import { useContext, useState } from 'react';
+import { GoogleMapsContext } from '../context/GoogleMapsContext';
+import { MapCenterContext } from '../context/MapCenterContext';
+import planetSearch from '../assets/search-location-solid.svg';
+
+export const SearchForm = () => {
+  const [query, setQuery] = useState('');
+  const valueGoogleMapsAPI = useContext(GoogleMapsContext);
+  const valueMapCenter = useContext(MapCenterContext);
+  const { map, maps } = valueGoogleMapsAPI;
+  const { setNewCenterMap } = valueMapCenter;
+
+  const handleQuery = (e) => setQuery(e.target.value);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const request = {
+      query,
+      fields: ['name', 'geometry'],
+    };
+
+    const service = new maps.places.PlacesService(map);
+
+    service.findPlaceFromQuery(request, function (results, status) {
+      if (status === maps.places.PlacesServiceStatus.REQUEST_DENIED) {
+        alert(
+          'We are so sorry! Places API is not enabled. Please try again later.',
+        );
+      }
+      if (status === maps.places.PlacesServiceStatus.ZERO_RESULTS) {
+        alert('No matching results');
+      }
+      if (status === maps.places.PlacesServiceStatus.OK) {
+        const firstResult = results[0];
+        const coordinates = firstResult?.geometry?.location;
```

Comment:
Not a big deal, but could you save a line by doing `const coordinates = results[0]?.geometry?.location;`?

Reply:
Saving lines! great! 😄